ambry-cloud/src/test/java/com.github.ambry.cloud/azure/AzureIntegrationTest.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.cloud.azure;
+
+import com.github.ambry.cloud.CloudBlobMetadata;
+import com.github.ambry.clustermap.MockPartitionId;
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.commons.BlobId;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.utils.TestUtils;
+import com.github.ambry.utils.Utils;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.Properties;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static com.github.ambry.commons.BlobId.*;
+import static org.junit.Assert.*;
+
+
+/**
+ * Integration Test cases for {@link AzureCloudDestination}
+ * Must be supplied with valid system property values for:
+ *   "storageConfigSpec" (Azure Blob Storage connection string)
+ *   "cosmosEndpoint"
+ *   "cosmosCollectionLink"
+ *   "cosmosKey"
+ */
+@RunWith(MockitoJUnitRunner.class)
+@Ignore
+public class AzureIntegrationTest {
+
+  private AzureCloudDestination azureDest;
+  private int blobSize = 1024;
+  private BlobId blobId;
+
+  @Before
+  public void setup() throws Exception {
+
+    byte dataCenterId = 66;
+    short accountId = 101;
+    short containerId = 5;
+    PartitionId partitionId = new MockPartitionId();
+    blobId = new BlobId(BLOB_ID_V6, BlobIdType.NATIVE, dataCenterId, accountId, containerId, partitionId, false,
+        BlobDataType.DATACHUNK);
+
+    VerifiableProperties verProps = new VerifiableProperties(System.getProperties());
+    azureDest = new AzureCloudDestination(verProps);
+  }
+
+  /**
+   * Test normal operations.
+   * @throws Exception
+   */
+  @Test
+  public void testNormalFlow() throws Exception {
```

Comment:
An overwrite test case would be helpful to make sure DB runs as expected.

Reply:
I'm not sure what you are asking for here.

Reply:
Do an overwrite to the same path after the first write, to make sure everything(DB and Storage) data are updated.

Reply:
Please ignore last comment. You introduced `testDeleteNotExists` and `testDeleteExists` in latest commit.

Reply:
I added an overwrite case in integration test.